src/components/browse/components/FacetList/index.js

```diff
@@ -337,64 +278,91 @@ export class FacetList extends React.PureComponent {
 
         const rgs = responsiveGridState(windowWidth || null);
 
-        function generateFacet(facet, i) {
-            const isStatic = Facet.isStatic(facet);
-            const defaultFacetOpen = !mounted ? false : !isStatic && !!(
-                ( rgs !== 'xs' && i < (facetIndexWherePastXTerms || 1) ) ||
-                (facet.aggregation_type === "stats" && _.any(filters || [], function(fltr){
-                    return (fltr.field === facet.field + ".from") || (fltr.field === facet.field + ".to");
-                })) ||
-                (facet.aggregation_type === "terms" && _.any(filters || [], function(fltr){
-                    return fltr.field === facet.field;
-                }))
-            );
-            return <Facet {...commonProps} facet={facet} key={facet.field} {...{ defaultFacetOpen, isStatic }} />;
-        }
+        // The logic within `Facet` `render`, `componentDidMount`, etc. isn't executed
+        // until is rendered by some other component's render method.
+        // We can sort/manipulate/transform these still according to their `props.` values and such.
+        const renderedFacets = useFacets.map(function(facet, i){
+            const {
+                grouping = null,
+                field: facetField,
+                aggregation_type = "terms"
+            } = facet;
+
+            // Default Open if mounted and:
+            let defaultFacetOpen = !mounted ? false : !!(rgs !== 'xs' && i < (facetIndexWherePastXTerms || 1));
+
+            if (aggregation_type === "stats") {
+                const { fromVal, toVal } = getRangeValueFromFilters(facet, filters);
+                const anySelected = fromVal !== null || toVal !== null;
+                const isStatic = facet.min === facet.max;
+                defaultFacetOpen = defaultFacetOpen || (!isStatic && _.any(filters || [], function(fltr){
+                    return (fltr.field === facetField + ".from") || (fltr.field === facetField + ".to");
+                })) || false;
+                return <RangeFacet {...commonProps} facet={facet} key={facetField} anyTermsSelected={anySelected} {...{ defaultFacetOpen, isStatic, grouping, fromVal, toVal }} />;
+            }
+            if (aggregation_type === "terms"){
+                const terms = mergeTerms(facet, filters); // Add in any terms specified in `filters` but not in `facet.terms` - in case someone hand-put that into URL.
+                const anySelected = anyTermsSelected(terms, facet, filters);
+                const isStatic = TermsFacet.isStatic(facet);
+                defaultFacetOpen = defaultFacetOpen || (!isStatic && _.any(filters || [], function(fltr){
```

Comment:
Maybe should remove the `!isStatic` check here. Even if static and selected, theoretically should be open. If is static then that prop might not be used (is 'always open'), but w.e...